syntax = "proto3";

package iptool;

// 描述網路模式；MANUAL 代表手動設定，DHCP 代表自動取得 IP
enum NetworkMode {
  NETWORK_MODE_UNSPECIFIED = 0;
  NETWORK_MODE_MANUAL = 1;
  NETWORK_MODE_DHCP = 2;
}

message GetNetworkConfigRequest {
  string interface_name = 1;
}

message NetworkConfig {
  string interface_name = 1;
  string ip_address = 2;
  string subnet_mask = 3;
  string gateway = 4;
  string dns = 5;
  NetworkMode mode = 6;
}

message GetNetworkConfigResponse {
  NetworkConfig config = 1;
}

message UpdateNetworkConfigRequest {
  NetworkConfig config = 1;
}

message UpdateNetworkConfigResponse {
  bool success = 1;
  string message = 2;
}

message SwitchToDhcpRequest {
  string interface_name = 1;
}

message SwitchToDhcpResponse {
  bool success = 1;
  string message = 2;
}

service NetworkService {
  rpc GetNetworkConfig(GetNetworkConfigRequest) returns (GetNetworkConfigResponse);
  rpc UpdateNetworkConfig(UpdateNetworkConfigRequest) returns (UpdateNetworkConfigResponse);
  rpc SwitchToDhcp(SwitchToDhcpRequest) returns (SwitchToDhcpResponse);
}
